onsite/lucxor/flr.py

```diff
@@ -1,798 +1,798 @@
+        
+        if delta_score > self.max_delta_score:
+            self.max_delta_score = delta_score
+            
+        if is_decoy:
+            self.decoy_psms.append(delta_score)
+            self.n_decoy += 1
+        else:
+            self.real_psms.append(delta_score)
+            self.n_real += 1
+            
+        logger.debug(f"Current real_psms count: {self.n_real}, decoy_psms count: {self.n_decoy}")
+    
+    def prep_arrays(self) -> None:
+        """Prepare arrays"""
+        self.pos = np.array(self.real_psms)
+        self.neg = np.array(self.decoy_psms)
+        self.n_real = len(self.pos)
+        self.n_decoy = len(self.neg)
+    
+    def initialize_tick_marks(self) -> None:
+        """Initialize tick marks"""
+        # First prepare arrays
+        self.prep_arrays()
+        
+        self.max_delta_score *= 1.001  # Need to be slightly larger for binning
+        
+        self.tick_marks = np.zeros(self.NMARKS)
+        for i in range(self.NMARKS):
+            x = (i * self.max_delta_score) / self.NMARKS
+            self.tick_marks[i] = x
+        
+        # Initialize other arrays
+        self.local_fdr = np.zeros(self.n_real)
+        self.global_fdr = np.zeros(self.n_real)
+        
+        self.calc_delta_score_mean()
+        self.calc_delta_score_var()
+        
+        self.get_bandwidth(DECOY)  # Decoy
+        self.get_bandwidth(REAL)   # Target
+        
+        # Output bandwidth information
+        logger.info(f"FLR bandwidth (pos): {self.bw_real:.6f}")
+        logger.info(f"FLR bandwidth (neg): {self.bw_decoy:.6f}")
+        
+
+    
+    def get_bandwidth(self, data_type: int) -> None:
+        """
+        Calculate bandwidth
+        
+        Args:
+            data_type: Data type (REAL or DECOY)
+        """
+        if data_type == REAL:
+            sigma = np.sqrt(self.delta_score_var_pos)
+            N = float(self.n_real)
+            x = np.power(N, 0.2)
+            result = 1.06 * (sigma / x)
+            self.bw_real = result
+            
+        elif data_type == DECOY:
+            sigma = np.sqrt(self.delta_score_var_neg)
+            N = float(self.n_decoy)
+            x = np.power(N, 0.2)
+            result = 1.06 * (sigma / x)
+            self.bw_decoy = result
+    
+    def calc_delta_score_mean(self) -> None:
+        """Calculate delta score mean"""
+        # Target sequence
+        sum_pos = 0.0
+        for d in self.pos:
+            sum_pos += d
+        N_pos = float(self.n_real)
+        self.delta_score_mu_pos = sum_pos / N_pos if N_pos > 0 else 0.0
+        
+        # Decoy sequence
+        sum_neg = 0.0
+        for d in self.neg:
+            sum_neg += d
+        N_neg = float(self.n_decoy)
+        self.delta_score_mu_neg = sum_neg / N_neg if N_neg > 0 else 0.0
+    
+    def calc_delta_score_var(self) -> None:
+        """Calculate delta score variance"""
+        # Target sequence
+        v = 0.0
+        for d in self.pos:

```

Comment:
Same here. np.var or at least vectorized operations for subtract and square to reuse mu.